feat: add organization authentication notes for Connected Accounts flow (#1041)
auth0/docs-v2: auth4genai/snippets/integrations/learn-more.mdx

## Learn more

- Auth0 stores the access and refresh tokens for each connected account in the [Token Vault](https://auth0.com/docs/secure/tokens/token-vault). Applications can then access the Token Vault to retrieve access tokens to call external APIs. To learn more, read [Configure Token Vault](https://auth0.com/docs/secure/tokens/token-vault/configure-token-vault).

> - If your application uses [Organizations](https://auth0.com/docs/manage-users/organizations), sign the user in to the correct organization before you initiate the Connected Accounts flow. Each user still connects their own external account.

Reviewer: ```suggestion
- If your application uses [Organizations](https://auth0.com/docs/manage-users/organizations), authenticate the user with the target organization before initiating the Connected Accounts flow. Each user still connects their own external account.
```
